### PR TITLE
[CARBONDATA-3889] Cleanup code typo in carbondata-spark module

### PR DESCRIPTION
**Pull request by QiangCai**
### Why is this PR needed?
 There are many typos in carbondata-spark module.
 
 ### What changes were proposed in this PR?
 Cleanup code typo in carbondata-spark module
    
 ### Does this PR introduce any user interface change?
 - No, not change interface name

 ### Is any new testcase added?
 - No, not impact function.

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3526/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1785/

File: integration/spark/src/main/scala/org/apache/carbondata/view/MVRefresher.scala [hunk 1]
```diff
@@ -68,19 +68,19 @@ object MVRefresher {
...
     val segmentStatusManager: SegmentStatusManager = new SegmentStatusManager(viewTableIdentifier)
-    // Acquire table status lock to handle concurrent dataloading
+    // Acquire table status lock to handle concurrent data loading
     val lock: ICarbonLock = segmentStatusManager.getTableStatusLock
     val segmentMapping: util.Map[String, util.List[String]] =
       new util.HashMap[String, util.List[String]]
     val viewManager = MVManagerInSpark.get(session)
     try if (lock.lockWithRetries) {
-      LOGGER.info("Acquired lock for mv " + viewIdentifier + " for table status updation")
+      LOGGER.info("Acquired lock for mv " + viewIdentifier + " for table status te")
```
Comment (kevinjmh) [inline integration/spark/src/main/scala/org/apache/carbondata/view/MVRefresher.scala:77]: check this

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/DeleteExecution.scala [hunk 2]
```diff
@@ -50,7 +50,7 @@ import org.apache.carbondata.events.{IndexServerLoadEvent, OperationContext, Ope
 import org.apache.carbondata.hadoop.api.{CarbonInputFormat, CarbonTableInputFormat}
 import org.apache.carbondata.processing.exception.MultipleMatchingException
 import org.apache.carbondata.processing.loading.FailureCauses
-import org.apache.carbondata.spark.DeleteDelataResultImpl
+import org.apache.carbondata.spark.DeleteDelateResultImpl
```
Comment (kevinjmh) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/DeleteExecution.scala:53]: delta

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/DeleteExecution.scala [hunk 3]
```diff
@@ -266,9 +266,9 @@ object DeleteExecution {
                 CarbonUpdateUtil.getRequiredFieldFromTID(TID, TupleIdEnum.BLOCK_ID) +
                 CarbonCommonConstants.FACT_FILE_EXT)
           }
-          val deleteDeletaPath = CarbonUpdateUtil
+          val deleteDeletePath = CarbonUpdateUtil
```
Comment (kevinjmh) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/DeleteExecution.scala:269]: ditto

File: integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/optimizer/CarbonSecondaryIndexOptimizer.scala [hunk 4]
```diff
@@ -693,10 +693,10 @@ class CarbonSecondaryIndexOptimizer(sparkSession: SparkSession) {
       case sort@Sort(order, global, plan) =>
         addProjection = true
         (sort, true)
-      case filter@Filter(condition, logicalRelation@MatchIndexableRelation(indexableRelation))
+      case filter@Filter(condition, logicalRelation@MatchIndexTableRelation(indexTableRelation))
```
Comment (kevinjmh) [inline integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/optimizer/CarbonSecondaryIndexOptimizer.scala:696]: rename is changing meaning. And actual code is to match CarbonDatasourceHadoopRelation

File: integration/spark/src/main/scala/org/apache/carbondata/view/MVRefresher.scala (hunk 1 above)
Comment (QiangCai) [reply integration/spark/src/main/scala/org/apache/carbondata/view/MVRefresher.scala]: done, change to 'update'

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/DeleteExecution.scala (hunk 2 above)
Comment (QiangCai) [reply integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/DeleteExecution.scala]: done

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/DeleteExecution.scala (hunk 3 above)
Comment (QiangCai) [reply integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/DeleteExecution.scala]: done, change to 'delta'

File: integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/optimizer/CarbonSecondaryIndexOptimizer.scala (hunk 4 above)
Comment (QiangCai) [reply integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/optimizer/CarbonSecondaryIndexOptimizer.scala]: Reverted the change.

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3552/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1812/

**Comment by kevinjmh** [comment]
LGTM

**Pull request closed without merging**